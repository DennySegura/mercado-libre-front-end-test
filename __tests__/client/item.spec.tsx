import 'jest';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Item from '../../src/app/components/result-list-item';

describe('Test suites for react components', () => {
  it('It should render <Item />', () => {
    const data ={
      title: 'Smart Tv 4k 50  Samsung Un50mu6100',
      price: {
        decimals: 0,
        amount: 21999,
        currency: 'ARS'
      },
      picture: 'http://mla-s1-p.mlstatic.com/849004-MLA27124959188_042018-I.jpg',
      free_shipping: false,
      location: 'Capital Federal'
    };
    const _item = renderer.create(<Item { ...data }/>);
    let item_tree = _item.toJSON();
    expect(item_tree).toMatchSnapshot();
  });
});
