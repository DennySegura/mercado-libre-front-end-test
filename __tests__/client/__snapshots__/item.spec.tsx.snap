// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suites for react components It should render <Item /> 1`] = `
<div
  className="item__container pv"
>
  <img
    alt="availability_image"
    className="item__image mr"
    src="http://mla-s1-p.mlstatic.com/849004-MLA27124959188_042018-I.jpg"
  />
  <div
    className="item__info--container"
  >
    <div
      className="item__description--container mt"
    >
      <span
        className="item__price--container mb3"
      >
        <span
          className="item__price--currency"
        >
          $
        </span>
        <span
          className="item__price--amount"
        >
          21.999
        </span>
      </span>
      <span
        className="item__description"
      >
        Smart Tv 4k 50  Samsung Un50mu6100
      </span>
    </div>
    <span
      className="caption tc-dusty-gray mt3 item__location"
    >
      Capital Federal
    </span>
  </div>
</div>
`;
